source/common/access_log/access_log_formatter.cc

```diff
@@ -136,10 +142,16 @@ std::vector<FormatterPtr> AccessLogFormatParser::parse(const std::string& format
 
         formatters.emplace_back(
             FormatterPtr(new ResponseHeaderFormatter(main_header, alternative_header, max_length)));
+      } else if (token.find("START_TIME") == 0) {
+        const size_t start = 11;
+
+        const std::string args = token[start - 1] == '('
```

Comment:
As a general, non-actionable comment for discussion, I think the header parsing logic is getting quite complicated to read/maintain/compose. I'd suggest we consider moving to some formal grammar and parser library going forward. @mattklein123 do you think this is worth opening an issue for?

Reply:
Agreed, I had suggested this as a prior "non-actionable" comment. I think we should just define a grammar and generate a parser using Bison, YACC, etc. (or whatever people use right now, it's been at least 15 years since I did anything in this area).

Reply:
https://github.com/envoyproxy/envoy/issues/2967

Reply:
@derekargueta can you slap a TODO for #2967 at some sensible place at the top of all this parsing logic? Thanks.